Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-52.ca-central-1.compute.internal
Sha: 0f3413e21b83b6c769e67bdbec9e6e08ee49722c
Build: success
Build: failed!
Number of Error(s): 12
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
\terrors: 0
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
\terrors: 0
CMake Error at /home/centos/smoketest/PR-14890/build/system/aws/cmake_install.cmake:49 (file):
CMake Error at /home/centos/smoketest/PR-14890/build/system/aws/cmake_install.cmake:56 (file):
CMake Error at /home/centos/smoketest/PR-14890/build/system/aws/cmake_install.cmake:63 (file):
CMake Error at /home/centos/smoketest/PR-14890/build/system/aws/cmake_install.cmake:70 (file):
CPack Error: Error when generating package: hpccsystems-platform
make: *** [Makefile:130: package] Error 1
CMake Error at /home/centos/smoketest/PR-14890/build/system/aws/cmake_install.cmake:49 (file):
  file RPATH_CHANGE could not write new RPATH:
    /opt/HPCCSystems/lib
  to the file:
    /home/centos/smoketest/PR-14890/build/system/aws/aws-sdk-cpp/aws-cpp-sdk-core/libaws-cpp-sdk-core.so
  No valid ELF RPATH or RUNPATH entry exists in the file;
Call Stack (most recent call first):
  /home/centos/smoketest/PR-14890/build/system/cmake_install.cmake:46 (include)
  /home/centos/smoketest/PR-14890/build/cmake_install.cmake:57 (include)
CMake Error at /home/centos/smoketest/PR-14890/build/system/aws/cmake_install.cmake:56 (file):
  file RPATH_CHANGE could not write new RPATH:
    /opt/HPCCSystems/lib
  to the file:
    /home/centos/smoketest/PR-14890/build/system/aws/aws-sdk-cpp/aws-cpp-sdk-sqs/libaws-cpp-sdk-sqs.so
  The current RPATH is:
    /home/centos/smoketest/PR-14890/build/system/aws/aws-sdk-cpp/aws-cpp-sdk-core:
  which does not contain:
    /home/centos/smoketest/PR-14890/build/system/aws/aws-sdk-cpp/aws-cpp-sdk-core:/home/centos/smoketest/PR-14890/build/system/aws/aws-sdk-cpp/.deps/install/lib64:
  as was expected.
Call Stack (most recent call first):
  /home/centos/smoketest/PR-14890/build/system/cmake_install.cmake:46 (include)
  /home/centos/smoketest/PR-14890/build/cmake_install.cmake:57 (include)
CMake Error at /home/centos/smoketest/PR-14890/build/system/aws/cmake_install.cmake:63 (file):
  file RPATH_CHANGE could not write new RPATH:
    /opt/HPCCSystems/lib
  to the file:
    /home/centos/smoketest/PR-14890/build/system/aws/aws-sdk-cpp/aws-cpp-sdk-s3/libaws-cpp-sdk-s3.so
  The current RPATH is:
    /home/centos/smoketest/PR-14890/build/system/aws/aws-sdk-cpp/aws-cpp-sdk-core:
  which does not contain:
    /home/centos/smoketest/PR-14890/build/system/aws/aws-sdk-cpp/aws-cpp-sdk-core:/home/centos/smoketest/PR-14890/build/system/aws/aws-sdk-cpp/.deps/install/lib64:
  as was expected.
Call Stack (most recent call first):
  /home/centos/smoketest/PR-14890/build/system/cmake_install.cmake:46 (include)
  /home/centos/smoketest/PR-14890/build/cmake_install.cmake:57 (include)
CMake Error at /home/centos/smoketest/PR-14890/build/system/aws/cmake_install.cmake:70 (file):
  file INSTALL cannot find
  /home/centos/smoketest/PR-14890/build/system/aws/aws-sdk-cpp/.deps/install/lib64/libaws-c-common.so.1.0.0.
Call Stack (most recent call first):
  /home/centos/smoketest/PR-14890/build/system/cmake_install.cmake:46 (include)
  /home/centos/smoketest/PR-14890/build/cmake_install.cmake:57 (include)
CPack Error: Error when generating package: hpccsystems-platform
make: *** [Makefile:130: package] Error 1
Package end (2021-04-23_14-18-19 5 sec
 ... ( comment is too long, 100 lines, 5663 bytes, truncated) ... 
[Build error:](https://gist.github.com/7744d1b709742ed921cfdf43d7804f17)
Build log [Build log](https://gist.github.com/HPCCSmoketest/7744d1b709742ed921cfdf43d7804f17#file-result-21-04-23-14-10-18-log)

```

--- a/esp/clients/wsdfuaccess/wsdfuaccess.cpp
+++ b/esp/clients/wsdfuaccess/wsdfuaccess.cpp
@@ -525,6 +525,8 @@ StringBuffer &encodeDFUFileMeta(StringBuffer &metaInfoBlob, IPropertyTree *metaI
 
         const char *keyPairName = metaInfo->queryProp("keyPairName");
         const char *privateKeyFName = environment->getPrivateKeyPath(keyPairName);
+        if (isEmptyString(privateKeyFName))
+            throw makeStringExceptionV(-1, "Key name '%s' is not found in environment settings: /EnvSettings/Keys/KeyPair.", keyPairName);
         Owned<CLoadedKey> privateKey = loadPrivateKeyFromFile(privateKeyFName, nullptr);
         StringBuffer metaInfoSignature;
         digiSign(metaInfoSignature, metaInfoBlob.length(), metaInfoBlob.bytes(), *privateKey);

--- a/esp/services/ws_dfu/ws_dfuService.cpp
+++ b/esp/services/ws_dfu/ws_dfuService.cpp
@@ -6030,6 +6030,10 @@ void CWsDfuEx::dFUFileAccessCommon(IEspContext &context, const CDfsLogicalFileNa
     unsigned port;
     bool secure;
     getFileDafilesrvConfiguration(keyPairName, port, secure, fileName, groups);
+#ifdef _USE_OPENSSL
+    if (secure && keyPairName.isEmpty())
+        throw makeStringExceptionV(-1, "No keyPairName is found for '%s' in environment settings: /EnvSettings/Keys/ClusterGroup.", cluster.str());
+#endif
 
     IEspDFUFileAccessInfo &accessInfo = resp.updateAccessInfo();
 

